src/app/data_harvester/processes/windows.rs

```diff
@@ -42,11 +41,11 @@ pub fn get_process_data(
         };
 
         let pcu = {
-            let p = process_val.cpu_usage() as f64 / num_processors;
-            if p.is_nan() {
-                process_val.cpu_usage() as f64
-            } else {
+            let p = process_val.cpu_usage() as f64;
+            if per_core_percentage || p.is_nan() {
                 p
+            } else {
+                p / (sys.cpus().len() as f64)
```

Comment:
The `p.is_nan()` check is mostly in the case of sysinfo reporting zero cores, thus causing a divide by zero. Since you've moved the division, it should be now like:

```rust
let p = process_val.cpu_usage() as f64;
if per_core_percentage {
  let res = p / (sys.cpus().len() as f64);
  if res.is_nan() {
    p
  } else {
    res
  }
}
else {
  p
}
```

Alternatively, you can just check if `sys.cpus().len()` is zero beforehand.